Portal_Perfomance_Employes/Controllers/UserInfoController.cs

```diff
@@ -0,0 +1,150 @@
+                }
+                else if (emp.Role == Role.Admin)
+                {
+                    return Ok(target);
+                }
+                else
+                {
+                    var empTeam = await _context.TeamMembers.FirstOrDefaultAsync(e => e.EmployeeId == empId && e.IsActive);
+                    var targetTeam = await _context.TeamMembers.FirstOrDefaultAsync(t => t.EmployeeId == targetId && t.IsActive);
+                    if (empTeam?.TeamId == null || targetTeam?.TeamId == null)
+                    {
+                        return StatusCode(403, "You do not have permission to view this profile");
+                    }
+                    else if (empTeam.TeamId == targetTeam.TeamId)
+                    {
+                        // Clearing targetTeam to not get fill "teamMemberships" value in response
+                        targetTeam = null;
+                        return Ok(target);
+                    } 
+                    return StatusCode(403, "You do not have permission to view this profile");
+                }
+            }
+            return StatusCode(403, "Error! Token is corrupt. ID of authorized user was not found");
+        }
+
+        [Route("GetUserColleagues")]
+        [HttpGet]
+        public async Task<IActionResult> GetUserColleagues()
+        {
+            int.TryParse(User.FindFirst(ClaimTypes.NameIdentifier)?.Value, out int id);
+            var emp = await _context.TeamMembers
+                .FirstOrDefaultAsync(e => e.EmployeeId == id && e.IsActive);
+            if (emp != null)
+            {
+                var colleagues = await _context.TeamMembers.Where(c => c.TeamId == emp.TeamId && c.IsActive).ToListAsync();
+                return colleagues == null ? NotFound("Something went wrong") : Ok(colleagues);
+            }
+            return NotFound("The team with that ID was not found");
+        }
+
+        [Route("GetUserTeamName")]
+        [HttpGet]
+        public async Task<IActionResult> GetUserTeamName()
+        {
+            int.TryParse(User.FindFirst(ClaimTypes.NameIdentifier)?.Value, out int id);
+            var emp = await _context.TeamMembers
+                .FirstOrDefaultAsync(e => e.EmployeeId == id && e.IsActive);
+            if (emp != null)
+            {
+                var team = await _context.Teams.FirstOrDefaultAsync(c => c.TeamId == emp.TeamId);
+                return team == null ? StatusCode(403, "Something went wrong") : Ok(team.TeamName);
+            }
+            return NotFound("Сurrent user is not assigned to any team");
+        }
+
+        [Route("GetUserTeamHistory")]
+        [HttpGet]
+        public async Task<IActionResult> GetUserTeamHistory()
+        {
+            int.TryParse(User.FindFirst(ClaimTypes.NameIdentifier)?.Value, out int id);
+            var emp = await _context.TeamMembers.Where(e => e.EmployeeId == id).ToListAsync();

```

Comment:
can't you get it from user itself?

Reply:
No, user has a ClaimsPrincipal type, and it is for authorization, so I have no possibility to get db context from it